File: edu-ios/Sources/Theme.swift
import SwiftUI

extension Color {
    init(hex: UInt) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xff) / 255,
            green: Double((hex >> 8) & 0xff) / 255,
            blue: Double(hex & 0xff) / 255
        )
    }
}

// Palette mirrors edu-web's globals.css: a dark, classical "lyceum" look —
// near-black crimson field, crimson surfaces, gold accents, parchment text.
enum Theme {
    // Raw palette
    static let crimson950 = Color(hex: 0x0f0404)
    static let crimson900 = Color(hex: 0x190808)
    static let crimson800 = Color(hex: 0x2d1212)
    static let crimson700 = Color(hex: 0x4a1a1a)
    static let gold500 = Color(hex: 0xb8860b)
    static let gold400 = Color(hex: 0xcfa135)
    static let gold300 = Color(hex: 0xddb954)

    // Semantic aliases (names kept stable across the app)
    static let parchment = crimson950 // screen background
    static let card = crimson900 // surfaces / cards
    static let parchmentDeep = crimson800 // secondary surfaces / selection
    static let line = crimson700 // borders / dividers
    static let ink = Color(hex: 0xf5ecd8) // primary text (parchment)
    static let inkSoft = Color(hex: 0xc4af8e) // secondary text (parchment-dim)
    static let crimson = gold300 // brand accent: titles, tints, selection
    static let gold = gold500 // gold accents (numbers, badges)
    static let accent = gold500 // primary-button fill
    static let onAccent = crimson950 // text on a gold fill
    static let success = Color(hex: 0x5cb85c)
    static let danger = Color(hex: 0xe06666)
    static let white = Color.white
}

// Brand fonts (bundled TTFs): Cinzel for display, EB Garamond for body — same as web.
extension Font {
    static func display(_ size: CGFloat) -> Font { .custom("Cinzel", size: size) }
    static func serif(_ size: CGFloat) -> Font { .custom("EB Garamond", size: size) }
}

extension View {
    /// Standard card surface used throughout the app.
    func lyceumCard() -> some View {
        self
            .padding(16)
            .background(Theme.card)
            .clipShape(RoundedRectangle(cornerRadius: 12))
            .overlay(RoundedRectangle(cornerRadius: 12).stroke(Theme.line, lineWidth: 1))
    }
}

// Press feedback for tappable cells/rows: a warm gold glow — a gold border, a faint gold
// wash, a soft outer glow, and a tiny press-in — so a selection reads clearly. Identical to
// `.plain` at rest (no chrome), lit only while the finger is down.
struct LyceumPressStyle: ButtonStyle {
    var cornerRadius: CGFloat = 12
    func makeBody(configuration: Configuration) -> some View {
        let down = configuration.isPressed
        return configuration.label
            .overlay(RoundedRectangle(cornerRadius: cornerRadius).fill(Theme.gold300.opacity(down ? 0.07 : 0)))
            .overlay(RoundedRectangle(cornerRadius: cornerRadius).strokeBorder(Theme.gold300, lineWidth: 1.5).opacity(down ? 1 : 0))
            .scaleEffect(down ? 0.985 : 1)
            .shadow(color: Theme.gold300.opacity(down ? 0.5 : 0), radius: down ? 12 : 0)
            // Snap to lit almost instantly on touch-down so the glow registers before a
            // quick tap triggers navigation; ease the fade-out a touch longer on release.
            .animation(down ? .easeOut(duration: 0.05) : .easeOut(duration: 0.22), value: down)
    }
}

extension ButtonStyle where Self == LyceumPressStyle {
    /// Gold press-glow for tappable cards/rows (drop-in replacement for `.plain`).
    static var lyceumPress: LyceumPressStyle { LyceumPressStyle() }
    static func lyceumPress(cornerRadius: CGFloat) -> LyceumPressStyle { LyceumPressStyle(cornerRadius: cornerRadius) }
}

File: edu-android/app/src/main/java/com/timpsonlyceum/lyceum/ui/theme/Theme.kt
package com.timpsonlyceum.lyceum.ui.theme

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Typography
import androidx.compose.material3.darkColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.timpsonlyceum.lyceum.R

// Palette mirrors edu-web's globals.css and the iOS app's Theme.swift — a dark,
// classical "lyceum" look: near-black crimson field, crimson surfaces, gold
// accents, parchment text. Names are kept identical to the iOS side on purpose,
// so a screen can be ported across without renaming a single colour.
object Theme {
    // Raw palette
    val crimson950 = Color(0xFF0F0404)
    val crimson900 = Color(0xFF190808)
    val crimson800 = Color(0xFF2D1212)
    val crimson700 = Color(0xFF4A1A1A)
    val gold500 = Color(0xFFB8860B)
    val gold400 = Color(0xFFCFA135)
    val gold300 = Color(0xFFDDB954)

    // Semantic aliases
    val parchment = crimson950      // screen background
    val card = crimson900           // surfaces / cards
    val parchmentDeep = crimson800  // secondary surfaces / selection
    val line = crimson700           // borders / dividers
    val ink = Color(0xFFF5ECD8)     // primary text
    val inkSoft = Color(0xFFC4AF8E) // secondary text
    val crimson = gold300           // brand accent
    val gold = gold500
    val accent = gold500
    val onAccent = crimson950
    val success = Color(0xFF5CB85C)
    val danger = Color(0xFFE06666)
}

// Brand fonts, the same TTFs the iOS app bundles: Cinzel for display, EB
// Garamond for body.
val Cinzel = FontFamily(Font(R.font.cinzel))
val EBGaramond = FontFamily(Font(R.font.eb_garamond))

/** Display face at an explicit size — the equivalent of iOS `.display(_:)`. */
fun display(size: Int) = TextStyle(fontFamily = Cinzel, fontSize = size.sp)

/** Body face at an explicit size — the equivalent of iOS `.serif(_:)`. */
fun serif(size: Int) = TextStyle(fontFamily = EBGaramond, fontSize = size.sp)

private val LyceumColors = darkColorScheme(
    primary = Theme.gold300,
    onPrimary = Theme.onAccent,
    secondary = Theme.gold500,
    background = Theme.parchment,
    onBackground = Theme.ink,
    surface = Theme.card,
    onSurface = Theme.ink,
    error = Theme.danger,
)

private val LyceumTypography = Typography(
    bodyLarge = TextStyle(fontFamily = EBGaramond, fontSize = 16.sp, color = Theme.ink),
    bodyMedium = TextStyle(fontFamily = EBGaramond, fontSize = 15.sp, color = Theme.ink),
    titleLarge = TextStyle(fontFamily = Cinzel, fontSize = 22.sp, color = Theme.ink),
    titleMedium = TextStyle(fontFamily = Cinzel, fontSize = 17.sp, color = Theme.ink),
    labelLarge = TextStyle(fontFamily = Cinzel, fontSize = 14.sp, color = Theme.ink),
)

@Composable
fun LyceumTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = LyceumColors,
        typography = LyceumTypography,
        content = content,
    )
}

/** Standard card surface used throughout the app — the `lyceumCard()` modifier. */
fun Modifier.lyceumCard(radius: Int = 12) = this
    .clip(RoundedCornerShape(radius.dp))
    .background(Theme.card)
    .border(1.dp, Theme.line, RoundedCornerShape(radius.dp))
    .padding(16.dp)
